pkg/agent/route/route.go

```diff
@@ -38,8 +39,19 @@ const (
 	AntreaIPRulePriority = 300
 )
 
+type Client interface {
+	Initialize(nodeConfig *config.NodeConfig) error
+	AddPeerCIDRRoute(peerPodCIDR *net.IPNet, gwLinkIdx int, peerNodeIP, peerGwIP net.IP) ([]*netlink.Route, error)
+	ListPeerCIDRRoute() (map[string][]*netlink.Route, error)
+	DeletePeerCIDRRoute(routes []*netlink.Route) error
+	RemoveServiceRouting() error
+	SetupPolicyOnlyMode() error
+	MigrateRoutesToGw(linkName string) error
+	UnMigrateRoutesFromGw(route *net.IPNet, linkName string) error
+}
```

Comment:
I have PR #462 doing similar change but hide more details from the interface's consumer to make windows implementation easier, can you take a look and see if it suits your PR too?

Reply:
will merge when PR#4762 is checked in